src/xeus_calc_interpreter.cpp

```
@@ -0,0 +1,178 @@
+                {
+                    ss << ops[s.top()] << ' ';
+                    s.pop();
+                }
+                s.pop();
+            }
+            else
+            {
+                ss << token << ' ';
+            }
+        }
+
+        while (!s.empty())
+        {
+            ss << ops[s.top()] << ' ';
+            s.pop();
+        }
+        publish_stream("stdout", "RPN = " + ss.str());
+        return ss.str();
+    }
+
+    double interpreter::compute_rpn(const std::string &expr)
+    {
+        publish_stream("stdout", "\nInput\tOperation\tStack after\n");
+        std::istringstream iss(expr);
+        std::vector<double> stack;
+        std::string token;
+        while (iss >> token)
+        {
+            publish_stream("stdout", token + "\t");
+            double tokenNum;
+            if (std::istringstream(token) >> tokenNum)
+            {
+                publish_stream("stdout","Push\t\t");
+                stack.push_back(tokenNum);
+            }
+            else
+            {
+                publish_stream("stdout", "Operate\t\t");
+                double secondOperand = stack.back();
+                stack.pop_back();
+                double firstOperand = stack.back();
+                stack.pop_back();
+                if (token == "*")
+                    stack.push_back(firstOperand * secondOperand);
+                else if (token == "/")
+                    stack.push_back(firstOperand / secondOperand);
+                else if (token == "-")
+                    stack.push_back(firstOperand - secondOperand);
+                else if (token == "+")
+                    stack.push_back(firstOperand + secondOperand);
+                else if (token == "^")
+                    stack.push_back(std::pow(firstOperand, secondOperand));
+                else
+                { //just in case
+                    publish_stream("stderr", "Error");
+                    std::exit(1);
```

Don't exit, publish an error message on IOPub and return. Or you can throw an exception that has to be catched in `execute_request_impl`